mw-webapp/src/component/viewSwitcher/ViewSwitcher.tsx

```diff
@@ -0,0 +1,95 @@
+import clsx from "clsx";
+import {HorizontalContainer} from "src/component/horizontalContainer/HorizontalContainer";
+import {Icon, IconProps, IconSize} from "src/component/icon/Icon";
+import {PositionTooltip} from "src/component/tooltip/PositionTooltip";
+import {Tooltip, TooltipProps} from "src/component/tooltip/Tooltip";
+import {View} from "src/utils/LocalStorageWorker";
+import styles from "src/component/viewSwitcher/ViewSwitcher.module.scss";
+
+/**
+ * View option
+ */
+export interface ViewOption {
+
+  /**
+   * View types
+   */
+  view: View;
+
+  /**
+   * Tooltip's content
+   */
+  tooltipContent: TooltipProps["content"];
+
+  /**
+   * Icon name
+   */
+  iconName: IconProps["name"];
+
+  /**
+   * Data attribute for cypress testing
+   */
+  dataCy?: string;
+}
+
+/**
+ * ViewSwitcher props
+ */
+export interface ViewSwitcherProps {
+
+  /**
+   * Custom class name
+   */
+  className?: string;
+
+  /**
+   * View value
+   */
+  view: View;
+
+  /**
+   * Set view value
+   */
+  setView: (view: View) => void;
+
+  /**
+   * Array of view options
+   */
+  options: ViewOption[];
+
+  /**
+   * Data attribute for cypress testing
+   */
+  dataCy?: string;
+}
+
+/**
+ * ViewSwitcher component
+ */
+export const ViewSwitcher = (props: ViewSwitcherProps) => {
+  return (
+    <HorizontalContainer
+      className={props.className}
+      dataCy={props.dataCy}
+    >
+      {props.options.map((option) => (
+        <Tooltip
+          key={option.view}
+          position={PositionTooltip.LEFT}
+          content={option.tooltipContent}
+          dataCy={option.dataCy}
+        >
+          <button
```

Comment:
We have component Button with type icon Button. Let's try to use component instead of default button if its possible)

Reply:
Done